ethcore/private-tx/src/lib.rs

```diff
@@ -599,6 +625,21 @@ impl Provider where {
 		let value = self.client.call_contract(block, *address, data)?;
 		decoder.decode(&value).map_err(|e| ErrorKind::Call(format!("Contract call failed {:?}", e)).into())
 	}
+
+	fn get_contract_version(&self, block: BlockId, address: &Address) -> Result<usize, Error> {
+		let (data, decoder) = private_contract::functions::get_version::call();
+		let value = self.client.call_contract(block, *address, data)?;
```

Comment:
Should we also fallback to `INITIAL_PRIVATE_CONTRACT_VER` if this call fails? (this way this function would return `usize` instead of `Result`).

Reply:
My suggestion is that we may want to fail early when possible. If someone deploys a contract that has illegal interface, then stop early. It wouldn't crash the client anyway!